port/aggregation-properties/resource_test.go

```diff
@@ -0,0 +1,787 @@
+	// The child blueprint has a relation to the parent blueprint.
+	// The parent blueprint has an aggregation property that counts the children of the parent.
+	// The aggregation property is created with a cycle relation to the child blueprint, which is allowed.
+	parentBlueprintIdentifier := utils.GenID()
+	childBlueprintIdentifier := utils.GenID()
+	var testAccActionConfigCreate = fmt.Sprintf(`
+	resource "port_blueprint" "parent_blueprint" {
+		title = "Parent Blueprint"
+		icon = "Terraform"
+		identifier = "%s"
+		description = ""
+		properties = {
+			number_props = {
+				"age" = {
+					title = "Age"
+				}
+			}
+		}
+	}
+
+	resource "port_blueprint" "child_blueprint" {
+		title = "Child Blueprint"
+		icon = "Terraform"
+		identifier = "%s"
+		description = ""
+       	relations = {
+            "parent" = {
+				 title = "Parent"
+				 target = port_blueprint.parent_blueprint.identifier
+			 }
+		}
+	}
+
+	resource "port_aggregation_properties" "child_aggregation_properties" {
+		blueprint_identifier = port_blueprint.parent_blueprint.identifier
+		properties = {
+			"count_entities" = {
+				target_blueprint_identifier = port_blueprint.child_blueprint.identifier
+				title = "Count Childrens"
+				icon = "Terraform"
+				description = "Count Childrens"
+				method = {	
+					count_entities = true	
+				}
+			}
+		}
+	}
+`, parentBlueprintIdentifier, childBlueprintIdentifier)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.TestAccPreCheck(t) },
+		ProtoV6ProviderFactories: acctest.TestAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: acctest.ProviderConfig + testAccActionConfigCreate,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("port_aggregation_properties.child_aggregation_properties", "blueprint_identifier", parentBlueprintIdentifier),
+					resource.TestCheckResourceAttr("port_aggregation_properties.child_aggregation_properties", "properties.count_entities.title", "Count Childrens"),
+					resource.TestCheckResourceAttr("port_aggregation_properties.child_aggregation_properties", "properties.count_entities.icon", "Terraform"),
+					resource.TestCheckResourceAttr("port_aggregation_properties.child_aggregation_properties", "properties.count_entities.description", "Count Childrens"),
+					resource.TestCheckResourceAttr("port_aggregation_properties.child_aggregation_properties", "properties.count_entities.target_blueprint_identifier", childBlueprintIdentifier),
+					resource.TestCheckResourceAttr("port_aggregation_properties.child_aggregation_properties", "properties.count_entities.method.count_entities", "true"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccCreateAggregationPropertyAverageEntities(t *testing.T) {
+	parentBlueprintIdentifier := utils.GenID()
+	childBlueprintIdentifier := utils.GenID()
+	var testAccActionConfigCreate = fmt.Sprintf(`
+	resource "port_blueprint" "parent_blueprint" {
```

Comment:
Maybe you can extract the parent and child blueprint definition and reuse, it will be more difficult to maintain it in the future like this